### PR TITLE
Update to xdebug 3 config and add ability to build app docker with xdebug

### LINKED ISSUE #1737: documentation about using docker with xdebug is out of date
The page 
https://github.com/wp-graphql/wp-graphql/blob/master/documentation/UsingDocker.md

references files which are no longer in the repository such as : `run-docker-local-app-xdebug.sh`.

Therefore, its no longer clear how to debug wp-graphql. It would be nice to update the documentation so that people can debug wp-graphql using docker.

### PR DESCRIPTION
**Pull request by markkelnar**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Allow building docker app and testing images with xdebub 3 installed and configured on port 9003. 

Does this close any currently open issues?
------------------------------------------
Fixes #1737 using xdebug in a docker with this repo.

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
Here is how I'd document this.
# Using Docker For Local Development

## WordPress Site

The `app` docker image starts a running WordPress site with the local wp-graphql directory installed and activated. Local changes to the source code is immediately reflects in the app.

1. Build the plugin. 
1.1 `composer install` or `docker run -v $PWD:/app composer --ignore-platform-reqs install`
1. Run `composer build-app` to build the `app` docker image.
1. Run `composer build-test` to build the `testing` docker image.
1. Run `composer run-app` to start the WordPress site.
1. Browse to http://localhost:8091/ to access the running WordPress app.
1. Browse to http://localhost:8091/wp-admin/ to access the admin dashboard. Username is 'admin'. Password is 'password'.

## Testing Suite

The `testing` docker image starts a running WordPress and runs the codeception test suites.

1. Run `composer build-test` to build the `testing` docker image.
1. Run `composer run-tests` to start the `testing` image and run the codeception tests.

# Using XDebug

## Local WordPress Site With XDebug
Use an environment variable USING_XDEBUG to start the docker image and WordPress with xdebug configured to use port 9003 to communicated with your IDE. 

```
export USING_XDEBUG=1
composer run-app
```

Start the debugger in your IDE. Set breakpoints. 

Load the app in http://localhost:8091/.

## Using XDebug With Tests

Use the environment variable USING_XDEBUG to run tests with xdebug configured to use port 9003 to communicated with your IDE. 

```
export USING_XDEBUG=1
composer run-tests
```

Start the debugger in your IDE. Set breakpoints.

## Configure VSCode IDE Launch File

Create or add the following configuration to your .vscode/launch.json in the root directory. Restart VSCode. Start the debug listener before running the app or testing images.

If you have WordPress core files in a directory for local development, you can add the location to the `pathMappings` for debug step through.

```
{
    "version": "0.2.0",
    "configurations": [
		{
			"name": "Listen for Xdebug",
			"type": "php",
			"request": "launch",
			"port": 9003,
			"xdebugSettings": {
				"max_children": 128,
				"max_data": 1024,
				"max_depth": 3,
				"show_hidden": 1
			},
			"pathMappings": {
				"/var/www/html/wp-content/plugins/wp-graphql": "${workspaceFolder}",
				"/var/www/html": "${workspaceFolder}/wordpress",
			}
		}
    ]
}
```

Where has this been tested?
---------------------------
**Operating System:** …

**WordPress Version:** …

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -54,7 +51,6 @@ services:
       DB_HOST: app_db
       WP_URL: 'http://localhost'
       WP_DOMAIN: 'localhost'
-      STRIPE_GATEWAY: 1
```
Comment (markkelnar) [inline docker-compose.yml:57]: I don't see where these are used in this repo.

Comment (jasonbahl) [reply docker-compose.yml:57]: Ya. . .they're not. Copy pasta from the docker config on the WPGraphQL for WooCommerce repo. 🤦‍♂️

Comment (markkelnar) [reply docker-compose.yml:57]: No worries.

**Comment by jasonbahl** [comment]
@markkelnar when I walk through this I am unable to run the app. 

When I followed the steps, I saw `WARNING: The USING_XDEBUG variable is not set. Defaulting to a blank string.`

So the steps need to change to the variable `USING_XDEBUG` instead of `USE_XDEBUG`.

When I ran with that changed, I still could not load the app at `http://localhost:8091/`. 

Screen Shot 2021-03-26 at 2 07 55 PM

Here's my full cli output after running `composer run-app`

```shell
➜  wp-graphql git:(fix/use-xdebug) composer run-app  
> bash bin/run-docker.sh run '-a'
Starting wp-graphql_app_db_1 ... done
Starting wp-graphql_app_1    ... done
Attaching to wp-graphql_app_db_1, wp-graphql_app_1
app_1      | Enabling XDebug 3
app_db_1   | 2021-03-26 20:05:06+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.7.33-1debian10 started.
app_1      | mv: cannot stat '/usr/local/etc/php/conf.d/disabled/docker-php-ext-xdebug.ini': No such file or directory
app_db_1   | 2021-03-26 20:05:06+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
app_db_1   | 2021-03-26 20:05:06+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.7.33-1debian10 started.
app_db_1   | 2021-03-26T20:05:06.588519Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
app_db_1   | 2021-03-26T20:05:06.589961Z 0 [Note] mysqld (mysqld 5.7.33) starting as process 1 ...
app_db_1   | 2021-03-26T20:05:06.593315Z 0 [Note] InnoDB: PUNCH HOLE support available
app_db_1   | 2021-03-26T20:05:06.593359Z 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
app_db_1   | 2021-03-26T20:05:06.593365Z 0 [Note] InnoDB: Uses event mutexes
app_db_1   | 2021-03-26T20:05:06.593372Z 0 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
app_db_1   | 2021-03-26T20:05:06.593380Z 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
app_db_1   | 2021-03-26T20:05:06.593397Z 0 [Note] InnoDB: Using Linux native AIO
app_db_1   | 2021-03-26T20:05:06.593698Z 0 [Note] InnoDB: Number of pools: 1
app_db_1   | 2021-03-26T20:05:06.593838Z 0 [Note] InnoDB: Using CPU crc32 instructions
app_db_1   | 2021-03-26T20:05:06.595361Z 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
app_db_1   | 2021-03-26T20:05:06.606423Z 0 [Note] InnoDB: Completed initialization of buffer pool
app_db_1   | 2021-03-26T20:05:06.608456Z 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
app_db_1   | 2021-03-26T20:05:06.620153Z 0 [Note] InnoDB: Highest supported file format is Barracuda.
app_db_1   | 2021-03-26T20:05:06.632377Z 0 [Note] InnoDB: Creating shared tablespace for temporary tables
app_db_1   | 2021-03-26T20:05:06.632467Z 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
app_db_1   | 2021-03-26T20:05:06.651141Z 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
app_1      | 2021/03/26 20:05:07 Waiting for: tcp://app_db:3306
app_db_1   | 2021-03-26T20:05:06.653312Z 0 [Note] InnoDB: 96 redo rollback segment(s) found. 96 redo rollback segment(s) are active.
app_db_1   | 2021-03-26T20:05:06.653326Z 0 [Note] InnoDB: 32 non-redo rollback segment(s) are active.
app_db_1   | 2021-03-26T20:05:06.654340Z 0 [Note] InnoDB: 5.7.33 started; log sequence number 39617849
app_db_1   | 2021-03-26T20:05:06.654617Z 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
app_db_1   | 2021-03-26T20:05:06.655004Z 0 [Note] Plugin 'FEDERATED' is disabled.
app_db_1   | 2021-03-26T20:05:06.657944Z 0 [Note] InnoDB: Buffer pool(s) load completed at 210326 20:05:06
app_db_1   | 2021-03-26T20:05:06.661433Z 0 [Note] Found ca.pem, server-cert.pem and server-key.pem in data directory. Trying to enable SSL support using them.
app_db_1   | 2021-03-26T20:05:06.661478Z 0 [Note] Skipping generation of SSL certificates as certificate files are present in data directory.
app_db_1   | 2021-03-26T20:05:06.662166Z 0 [Warning] CA certificate ca.pem is self signed.
app_db_1   | 2021-03-26T20:05:06.662210Z 0 [Note] Skipping generation of RSA key pair as key files are present in data directory.
app_db_1   | 2021-03-26T20:05:06.662835Z 0 [Note] Server hostname (bind-address): '*'; port: 3306
app_db_1   | 2021-03-26T20:05:06.662969Z 0 [Note] IPv6 is available.
app_db_1   | 2021-03-26T20:05:06.662996Z 0 [Note]   - '::' resolves to '::';
app_db_1   | 2021-03-26T20:05:06.663014Z 0 [Note] Server socket created on IP: '::'.
app_db_1   | 2021-03-26T20:05:06.664785Z 0 [Warning] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
app_db_1   | 2021-03-26T20:05:06.682070Z 0 [Note] Event Scheduler: Loaded 0 events
app_db_1   | 2021-03-26T20:05:06.682642Z 0 [Note] mysqld: ready for connections.
app_db_1   | Version: '5.7.33'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server (GPL)
app_1      | 2021/03/26 20:05:07 Connected to tcp://app_db:3306
app_db_1   | 2021-03-26T20:05:07.539365Z 2 [Note] Got an error reading communication packets

```

**Comment by markkelnar** [comment]
> ```shell
> mv: cannot stat '/usr/local/etc/php/conf.d/disabled/docker-php-ext-xdebug.ini': No such file or directory
> ```

This error is the xdebug config to enable it.  So, something was mismatch or missing.  Try building app and running again.

**Pull request merged**